src/field_theory/minimal_polynomial.lean

```diff
@@ -0,0 +1,230 @@
+/-
+Copyright (c) 2019 Chris Hughes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Hughes, Johan Commelin
+-/
+
+import ring_theory.integral_closure
+
+/-!
+# Minimal polynomials
+
+This file defines the minimal polynomial of an element x of an A-algebra B,
+under the assumption that x is integral over A.
+
+After stating the defining property we specialize to the setting of field extensions
+and derive some well-known properties, amongst which the fact that minimal polynomials
+are irreducible, and uniquely determined by their defining property.
+
+-/
+
+universes u v w
+
+-- local attribute [instance, priority 1] classical.prop_decidable
+-- open_locale classical
+
+open polynomial set function
+
+variables {α : Type u} {β : Type v}
+
+section min_poly_def
+variables [decidable_eq α] [decidable_eq β] [comm_ring α] [comm_ring β] [algebra α β]
+
+/-- Let B be an A-algebra, and x an element of B that is integral over A.
+The minimal polynomial of x if the monic polynomial of smallest degree that has x as its root. -/
+noncomputable def minimal_polynomial {x : β} (hx : is_integral α x) : polynomial α :=
+well_founded.min polynomial.degree_lt_wf _ (ne_empty_iff_exists_mem.mpr hx)
+
+end min_poly_def
```

Comment:
"if the monic" -> "is the monic" in docstring. But "the" implies uniqueness. If $A=\Z$ and $B=\Z[X]/(X^2,2X)=\Z\iplus\F_2\epsilon$ I think both $X^2$ and $X^2-2X$ are min polys for $\epsilon$.

Hmm. How about adding something like "unique if B is an integral domain" or something?